services/frontend-service/src/ui/components/ProductVersion/ProductVersion.test.tsx

```diff
@@ -13,30 +13,46 @@ You should have received a copy of the MIT License
 along with kuberpult. If not, see <https://directory.fsf.org/wiki/License:Expat>.
 
 Copyright 2023 freiheit.com*/
+import { screen, render } from '@testing-library/react';
 import { MemoryRouter } from 'react-router-dom';
+import { GetGitTagsResponse, TagData } from '../../../api/api';
+import { updateTag } from '../../utils/store';
 import { ProductVersion } from './ProductVersion';
-import { render } from '@testing-library/react';
 
 describe('Product Version Data', () => {
     type TestData = {
         name: string;
         environmentName: string;
+        expectedDropDown: string;
+        tags: TagData[];
     };
     const data: TestData[] = [
         {
             name: 'No tags to Display',
             environmentName: 'tester',
+            tags: [],
+            expectedDropDown: 'Select a Tag',
+        },
+        {
+            name: 'tags to Display',
+            environmentName: 'tester2',
+            tags: [{ commitId: '123', tag: 'refs/tags/dummyTag' }],
+            expectedDropDown: 'dummyTag',
         },
     ];
 
     describe.each(data)(`Displays Product Version Page`, (testCase) => {
+        // given
         it(testCase.name, () => {
+            var tagsResponse: GetGitTagsResponse = { tagData: testCase.tags };
+            updateTag.set(tagsResponse);
             render(
                 <MemoryRouter>
                     <ProductVersion environment={testCase.environmentName} />
                 </MemoryRouter>
             );
             expect(document.querySelector('.environment_name')?.textContent).toContain(testCase.environmentName);
+            expect(screen.getByText(testCase.expectedDropDown)).toBeInTheDocument();
```

Comment:
please use a more precise way of testing here, like you do in the line before. `getByText` is okay for a first trying things out, but then we need to replace it with something that uses a selector.
Something like this:
```suggestion
            expect(document.querySelector('.myclass .tag123....')?.textContent).toContain(testCase.expectedTagOption);
```